habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py

```diff
@@ -0,0 +1,248 @@
+        self._active_agents = np.random.choice(
+            self._all_agent_idxs,
+            size=self._pop_config.num_active_agents,
+            replace=self._pop_config.allow_self_play,
+        )
+
+        self._multi_storage.set_active(
+            [self._agents[i].rollouts for i in self._active_agents]
+        )
+        self._multi_updater.set_active(
+            [self._agents[i].updater for i in self._active_agents]
+        )
+        self._multi_policy.set_active(
+            [self._agents[i].actor_critic for i in self._active_agents]
+        )
+
+    def post_init(self, create_rollouts_fn=None):
+        for agent in self._agents:
+            agent.post_init(create_rollouts_fn)
+
+        self._num_updates = 0
+        if not self._pop_config.self_play_batched:
+            self._sample_active()
+
+    def eval(self):
+        for agent in self._agents:
+            agent.eval()
+
+    def train(self):
+        for agent in self._agents:
+            agent.train()
+
+    @property
+    def num_total_agents(self):
+        return len(self._agents)
+
+    def load_state_dict(self, state):
+        for agent in self._agents:
+            agent.load_state_dict(state)
+
+    def load_ckpt_state_dict(self, ckpt):
+        for agent in self._agents:
+            agent.load_ckpt_state_dict(ckpt)
+
+    @property
+    def hidden_state_shape(self):
+        """
+        Stack the hidden states of all the policies in the active population.
+        """
+
+        return np.sum(
+            np.stack(
+                [
+                    self._agents[i].hidden_state_shape
+                    for i in self._active_agents
+                ]
+            ),
+            axis=0,
+        )
+
+    def after_update(self):
+        """
+        Will resample the active agent population every `agent_sample_interval` calls to this method.
+        """
+
+        for agent in self._agents:
+            agent.after_update()
+        self._num_updates += 1
+        if (
+            self._num_updates % self._pop_config.agent_sample_interval == 0
+            and self._pop_config.agent_sample_interval != -1
+        ):
+            self._sample_active()
+
+    def pre_rollout(self):
+        for agent in self._agents:
+            agent.pre_rollout()
+
+    def get_resume_state(self):
+        return {
+            str(agent_i): agent.get_resume_state()
+            for agent_i, agent in enumerate(self._agents)
+        }
+
+    def get_save_state(self):
+        return {
+            agent_i: agent.get_save_state()
+            for agent_i, agent in enumerate(self._agents)
+        }
+
+    @property
+    def rollouts(self):
+        return self._multi_storage
+
+    @property
+    def actor_critic(self):
+        return self._multi_policy
+
+    @property
+    def updater(self):
+        return self._multi_updater
+
+    @property
+    def policy_action_space(self):
+        return self._agents[0].policy_action_space
```

Comment:
Why only of the first agent? Isn't it supposed to be combined ?

Reply:
Oops, that was not intentional. Fixed.